integration.test.js

```diff
@@ -0,0 +1,111 @@
+/* global describe, beforeAll, afterAll, it, expect */
+require('dotenv').config()
+const Fastify = require('fastify')
+const fetch = require('cross-fetch')
+
+if (
+  !process.env.AUTH0_DOMAIN ||
+  !process.env.AUTH0_CLIENT_ID ||
+  !process.env.AUTH0_CLIENT_SECRET ||
+  !process.env.AUTH0_API_AUDIENCE
+) {
+  throw new Error('Integration tests needs a set of environment variables to be set')
+}
+
+async function buildServer() {
+  const server = Fastify()
+
+  // Setup fastify-auth0-verify
+  server.register(require('.'), {
+    domain: process.env.AUTH0_DOMAIN,
+    secret: process.env.AUTH0_CLIENT_SECRET
+  })
+
+  // Setup auth0 protected route
+  server.register(async function protectedRoute(fastify) {
+    server.get('/protected', { preValidation: [server.authenticate] }, async (req, reply) => {
+      reply.send({ route: 'Protected route' })
+    })
+  })
+
+  // Setup auth0 public route
+  server.register(async function publicRoute(fastify) {
+    server.get('/public', async (req, reply) => {
+      reply.send({ route: 'Public route' })
+    })
+  })
```

Comment:
same as before. why this registration? there's no need. just define the routes inline